packages/trpc/server/routers/viewer.tsx

```diff
@@ -1138,10 +1140,58 @@ const loggedInViewerRouter = router({
     });
     return recurringGrouping.reduce((prev, current) => {
       // recurringEventId is the total number of recurring instances for a booking
-      // we need to substract all but one, to represent a single recurring booking
+      // we need to subtract all but one, to represent a single recurring booking
       return prev - (current._count?.recurringEventId - 1);
     }, count);
   }),
+  getCalVideoRecordings: authedProcedure
+    .input(
+      z.object({
+        roomName: z.string(),
+      })
+    )
+    .query(async ({ ctx, input }) => {
+      const { roomName } = input;
+      let shouldHideRecordingsData = false;
+      // If self-hosted, he should be able to get recordings
+      if (IS_TEAM_BILLING_ENABLED) {
```

Comment:
@Udit-takkar Added server validation to don't leak any room info on client if user has no team plan.

Reply:
i don't think not passing room name or id is of any use here because for getting any data like download link you need correct api key. so event if you get access to the room id you can't do much with it.

Reply:
this is also giving some type errors in ViewRecordingsDialog for some reason